### PR TITLE
Add websocket JS tests and call onclose on failure

### PR DESCRIPTION
**Pull request by dralletje**

### CODE REVIEW

**Comment by dralletje** [comment]
Made this pull request instead of #3186 because I didn't know how to rebase it correctly, sorry!
But I added JS tests here, not sure if that is done correctly though ( @cpojer )

**Comment by dralletje** [comment]
And just for more feedback :grin: Is this what you were thinking of in #2984 @ide ?

**Comment by satya164** [comment]
This is going to have conflicts after #2893 gets merged.

**Comment by satya164** [comment]
This is going to have conflicts after #2837 gets merged.

**Comment by dralletje** [comment]
@satya164 Not really conflict, but indeed because of the WebSocketBase it'll cause problems.
Do you have time tomorrow anywhere to resolve the problems together? (I didn't get the tests to run for android just yet, so we could look at that too maybe)

**Comment by satya164** [comment]
OK. I'll try.

File: Libraries/WebSocket/__tests__/Websocket-test.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+jest
+  .autoMockOff()
```
Comment (cpojer) [inline Libraries/WebSocket/__tests__/Websocket-test.js]: why do you have to turn off auto-mocking? Can't you just remove this line and continue mocking `NativeModules` just like you do below?

Comment (dralletje) [inline Libraries/WebSocket/__tests__/Websocket-test.js]: @cpojer I wish I could, but I couldn't make that work (ran into all kinds of errors, even when I was dontMock'ing all the things I needed). If you want to have a try, go ahead :)

**Comment by christopherdro** [comment]
Related #4802

**Comment by facebook-github-bot-9** [comment]
@vjeux would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
@dralletje It looks like this needs rebasing now. Do you plan to continue working on this?

**Comment by mkonicek** [comment]
cc @satya164

**Comment by satya164** [comment]
@dralletje Can you rebase against master?

**Comment by facebook-github-bot-6** [comment]
@AaaChiuuu would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
Closing this, if you want to continue working on it please create a new pull request that only contains the changes you intend to make. This one has over 1000 changed files.

**Comment by dralletje** [comment]
Yeah, I dont know how I screwed this thing up, hahahaha

**Pull request closed without merging**